paper-autocomplete.html

```diff
@@ -65,158 +62,180 @@
       --paper-item-min-height: 36px;
     }
   </style>
-
   <template>
     <div class="input-wrapper">
-      <paper-input id="input" label="[[label]]" no-label-float="[[noLabelFloat]]" on-keyup="_onKeypress" disabled="{{disabled}}" auto-validate$="[[autoValidate]]" error-message$="[[errorMessage]]" required$="[[required]]" value="{{text}}"></paper-input>
-      <paper-icon-button id="clear" icon="clear" on-click="_clear"></paper-icon-button>
+      <paper-input id="input" label="[[label]]" allowed-pattern="[[allowedPattern]]" pattern="[[pattern]]" no-label-float="[[noLabelFloat]]" on-keyup="_onKeypress" disabled="{{disabled}}" auto-validate$="[[autoValidate]]" error-message$="[[errorMessage]]" required$="[[required]]" value="{{text}}">
+      </paper-input>
+      <paper-icon-button id="clear" icon="clear" on-click="_clear">
+      </paper-icon-button>
     </div>
     <paper-material elevation="1" id="suggestionsWrapper">
       <template is="dom-repeat" items="{{_suggestions}}">
         <paper-item on-click="_onSelect">
-          <div>{{item.text}}</div>
-          <paper-ripple></paper-ripple>
+          <div>{{item.text}}
+          </div>
+          <paper-ripple>
+          </paper-ripple>
         </paper-item>
       </template>
     </paper-material>
   </template>
 </dom-module>
-
 <script>
   Polymer({
     is: 'paper-autocomplete',
     properties: {
       /**
-       * `autoValidate` Set to true to auto-validate the input value.
-       */
+     * `autoValidate` Set to true to auto-validate the input value.
+     */
       autoValidate: {
         type: Boolean,
         value: false
-      },
-
+      }
+      ,
       /**
-       * `errorMessage` The error message to display when the input is invalid.
-       */
+     * `errorMessage` The error message to display when the input is invalid.
+     */
       errorMessage: {
         type: String,
-      },
+      }
+      ,
       /**
-       * `label` Text to display as the input label
-       */
+     * `label` Text to display as the input label
+     */
       label: String,
       /**
-       * `noLabelFloat` Set to true to disable the floating label.
-       */
+     * `pattern` Pattern to validate input field
+     */
+      pattern:String,
+      /**
+     * `allowedPattern` allowedPattern to validate input field
+     */
+      allowedPattern:String,
+      /**
+     * `noLabelFloat` Set to true to disable the floating label.
+     */
       noLabelFloat:{
         type: Boolean,
         value: false
-      },
-
-       /**
-       * `required` Set to true to mark the input as required.
-       */
+      }
+      ,
```

Comment:
I would be careful with formatting the code. The list of changes in this PR is pretty big, but most of the changes are only style changes. I guess the maintainers will have something to say about this.